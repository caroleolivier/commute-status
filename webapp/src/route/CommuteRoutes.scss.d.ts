export const container: string;
export const header: string;
export const content: string;
export const footer: string;
