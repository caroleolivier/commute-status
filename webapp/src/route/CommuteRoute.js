import React, { Component } from 'react';
import PropTypes from 'prop-types';

import BusStationContainer from '../bus/BusStationContainer';
import TubeStationContainer from '../tube/TubeStationContainer';

const CommuteRouteStopType = {
    BUS: {
        getComponent: stopConfig => <BusStationContainer config={stopConfig} />
    },
    TUBE: {
        getComponent: stopConfig => <TubeStationContainer config={stopConfig} />
    }
};

class CommuteRoute extends Component {
    render() {
        const stops = this.props.stops.map(stopConfig =>
            CommuteRouteStopType[stopConfig.type.toUpperCase()].getComponent(stopConfig));
        return (
            <div>
                <h2>{this.props.routeName}</h2>
                {stops}
            </div>
        );
    }
}

CommuteRoute.propTypes = {
    routeName: PropTypes.string.isRequired,
    stops: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default CommuteRoute;
