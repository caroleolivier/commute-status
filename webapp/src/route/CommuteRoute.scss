.commuteRoute {
    padding: 10px;
}