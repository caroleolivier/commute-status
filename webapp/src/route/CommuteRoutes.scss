.container {
    height: 100%;
}

.header {
    height: 15%;
}

.content {
    height: 75%;
}

.footer {
    height: 10%;
    display: flex;
    flex-direction: column;
    justify-content: center;
}