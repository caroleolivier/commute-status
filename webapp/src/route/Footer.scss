.footer {
    display: flex;
    justify-content: center;

    > a {
        color: white;
        padding: 0px 3px;
    }
}