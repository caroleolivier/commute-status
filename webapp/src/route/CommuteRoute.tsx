import * as React from 'react';

import { StationContainer } from './station/StationContainer';
import * as styles from './CommuteRoute.scss';

export interface ICommuteRouteConfig {
    routeName: string;
    stops: any[];
}

interface ICommuteRouteProps {
    config: ICommuteRouteConfig;
}

export class CommuteRoute extends React.Component<ICommuteRouteProps, {}> {
    render(): JSX.Element {
        return (
            <div className={styles.commuteRoute}>
                <h2>{this.props.config.routeName}</h2>
                <div className="horizontalContainer">
                    {this.props.config.stops.map(stopConfig => <StationContainer key={stopConfig.stationId} config={stopConfig} />)}
                </div>
            </div>
        );
    }
}
