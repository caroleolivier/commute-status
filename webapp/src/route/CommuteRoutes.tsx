import * as React from 'react';

import * as styles from './CommuteRoutes.scss';
import { ICommuteRoutesConfig } from '../config/configtype';
import { Header } from './Header';
import { Footer } from './Footer';
import { CommuteRoute } from './CommuteRoute';

interface ICommuteRoutesProps {
    config: ICommuteRoutesConfig;
}

export class CommuteRoutes extends React.Component<ICommuteRoutesProps, {}> {
    render(): JSX.Element {
        const routesComponents = this.props.config.routes.map(
            routeConfig => <CommuteRoute key={routeConfig.routeName} config={routeConfig} />);
        return (
            <div className={styles.container}>
                <div className={styles.header} >
                    <Header name={this.props.config.name} />
                </div>
                <div className={styles.content}>
                    {routesComponents}
                </div>
                <div className={styles.footer} >
                    <Footer/>
                </div>
            </div>
        );
    }
}