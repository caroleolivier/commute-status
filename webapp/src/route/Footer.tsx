import * as React from 'react';
import * as styles from './Footer.scss';

export class Footer extends React.Component {
    render() {
        return (
            <div className={styles.footer}>
                Data from <a href="https://api.tfl.gov.uk/">TfL</a> and <a href="http://www.nationalrail.co.uk/46391.aspx">NRE</a>
            </div>
        );
    }
}