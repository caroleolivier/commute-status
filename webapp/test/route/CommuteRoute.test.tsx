import 'jest';
import * as React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { CommuteRoute, ICommuteRouteConfig } from '../../src/route/CommuteRoute';

describe('CommuteRoute', () => {
    describe('Given route name and a list of stops', () => {
        const config: ICommuteRouteConfig = {
            routeName: 'Via the Moon',
            stops: [
                {
                    type: 'bus',
                    stationName: 'River',
                    stationId: '2erwerq',
                    travelDirection: 'Saturne'
                },
                {
                    type: 'tube',
                    stationName: 'Sea',
                    stationId: 'qwerds',
                    travelDirection: 'Mars'
                }
            ]
        };

        test('it creates the correct stop components', () => {
            let commuteRouteWrapper = shallow(<CommuteRoute config={config} />);
            const tree = toJson(commuteRouteWrapper);
            expect(tree).toMatchSnapshot('routes');
        });
    });
});
