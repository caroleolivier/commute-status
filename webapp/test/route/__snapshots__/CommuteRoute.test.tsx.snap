// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes 1`] = `
<div
  className="commuteRoute"
>
  <h2>
    Via the Moon
  </h2>
  <div
    className="horizontalContainer"
  >
    <StationContainer
      config={
        Object {
          "stationId": "2erwerq",
          "stationName": "River",
          "travelDirection": "Saturne",
          "type": "bus",
        }
      }
      key="2erwerq"
    />
    <StationContainer
      config={
        Object {
          "stationId": "qwerds",
          "stationName": "Sea",
          "travelDirection": "Mars",
          "type": "tube",
        }
      }
      key="qwerds"
    />
  </div>
</div>
`;
